packages/datadog-esbuild/index.js

```diff
@@ -2,73 +2,111 @@
 
 /* eslint-disable no-console */
 
-const NAMESPACE = 'datadog'
-
-const instrumented = Object.keys(require('../datadog-instrumentations/src/helpers/hooks.js'))
-const rawBuiltins = require('module').builtinModules
+const path = require('path')
+const fs = require('fs')
+const instrumentations = require('../datadog-instrumentations/src/helpers/instrumentations.js')
 
 warnIfUnsupported()
 
+{
+  // run each instrumentation so that `instrumentations` gets populated
+  const INS_PATH = path.join(__dirname, '../datadog-instrumentations/src')
+
+  for (const entry of fs.readdirSync(INS_PATH)) {
+    if (path.extname(entry) !== '.js') continue
+    require(path.join(INS_PATH, entry))
+  }
+}
+
+const modulesOfInterest = new Set()
+
+for (const instrumentation of Object.values(instrumentations)) {
+  for (const entry of instrumentation) {
+    if (!entry.file) {
+      modulesOfInterest.add(entry.name) // redis
```

Comment:
This comment is confusing. It took me a while to understand that this has nothing to do with redis.

Reply:
I fixed up the two comments, they should be more self-explanatory now.